packages/module/src/Hooks/sort.ts

```diff
@@ -0,0 +1,87 @@
+import { ISortBy } from "@patternfly/react-table";
+import { useState, useEffect, useMemo } from "react";
+
+export enum DataViewSortParams {
+  SORT_BY = 'sortBy',
+  DIRECTION = 'direction'
+};
+
+const validateDirection = (direction: string | null | undefined, defaultDirection: ISortBy['direction']): ISortBy['direction'] => (
+  direction === 'asc' || direction === 'desc' ? direction : defaultDirection
+);
+
+export interface DataViewSortConfig {
+  /** Attribute to sort the entries by */
+  sortBy: string | undefined;
+  /** Sort direction */
+  direction: ISortBy['direction'];
+};
+
+export interface UseDataViewSortProps {
+  /** Initial sort config */
+  initialSort?: DataViewSortConfig;
+  /** Current search parameters as a string */
+  searchParams?: URLSearchParams;
+  /** Function to set search parameters */
+  setSearchParams?: (params: URLSearchParams) => void;
+  /** Default direction */
+  defaultDirection?: ISortBy['direction'];
+  /** Sort by URL param name */
+  sortByParam?: string;
+  /** Direction URL param name */
+  directionParam?: string;
+};
+
+export const useDataViewSort = (props?: UseDataViewSortProps) => {
+  const {
+    initialSort,
+    searchParams,
+    setSearchParams,
+    defaultDirection = 'asc',
+    sortByParam = DataViewSortParams.SORT_BY,
+    directionParam = DataViewSortParams.DIRECTION
+  } = props ?? {};
+
+  const isUrlSyncEnabled = useMemo(() => searchParams && !!setSearchParams, [ searchParams, setSearchParams ]);
+
+  const [ state, setState ] = useState<DataViewSortConfig>({
+    sortBy: searchParams?.get(sortByParam) ?? initialSort?.sortBy,
+    direction: validateDirection(searchParams?.get(directionParam) as ISortBy['direction'], initialSort?.direction),
+  });
+
+  const updateSearchParams = (sortBy: string, direction: ISortBy['direction']) => {
+    if (isUrlSyncEnabled && sortBy) {
+      const params = new URLSearchParams(searchParams);
+      params.set(sortByParam, `${sortBy}`);
+      params.set(directionParam, `${direction}`);
+      setSearchParams?.(params);
+    }
+  };
+
+  useEffect(() => {
+    state.sortBy && state.direction && updateSearchParams(state.sortBy, state.direction);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const currentSortBy = searchParams?.get(sortByParam) || state.sortBy;
+    const currentDirection = searchParams?.get(directionParam) as ISortBy['direction'] || state.direction;
+    const validDirection = validateDirection(currentDirection, defaultDirection);
+    currentSortBy !== state.sortBy || validDirection !== state.direction && setState({ sortBy: currentSortBy, direction: validDirection });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ searchParams?.toString() ]);
```

Comment:
This is too broad IMO. We should only look at the sort params. Your effect will trigger on any search param change. So for pagination for example. Which is not needed.